### ISSUE TITLE
HeuristicDriveShaper normalizes the pulse differently than the register

### ISSUE THREAD

**Issue by CdeTerra** [pasqal-io/qubo-solver#198]
### Short description

The pulse's values should proportionally correspond to the involved QUBO at the embedding scaling, for any embedder, so that it is adapted to the register.

The fix is very likely in `HeuristicDriveShaper.generate` on the line `Q = self.qubo_normalized_coefficients` that uses normalized coefficients independently from the embedding. I'll be working on it.

### What is the expected result?

Using a toy example of size 2, linear terms set to $-1$ and a variable interaction between the 2 atoms, the ratio $\frac{\text{embedding interaction}}{\text{drive detuning}}$ should be equal to $\frac{\text{qubo quadratic}}{\text{qubo linear}}$, multiplied by $-1$ (detuning convention) and by $2$ (QUBO solver convention because the matrix is symmetrical)

### What is the actual result?

The ratio is only correct using BLaDE with the quadratic term set to 1.

### Steps/Code to reproduce

```python
import torch
from qubosolver import QUBOInstance
from qubosolver.config import SolverConfig, EmbeddingConfig
from qubosolver.solver import QuboSolver

from qoolqit.graphs import DataGraph

from scipy.spatial.distance import pdist, squareform
import numpy as np
import networkx as nx


def embedding_drive_scalings(embedding_method: str, interaction: float = 1):
    qubo = np.array([
        [-1, interaction],
        [interaction, -1],
    ])

    instance = QUBOInstance(coefficients=qubo)

    # Create a SolverConfig object to use a quantum backend
    config = SolverConfig(use_quantum=True, embedding=EmbeddingConfig(embedding_method=embedding_method))

    # Instantiate the quantum solver.
    solver = QuboSolver(instance, config)

    # Solve the QUBO problem.
    solution = solver.solve()
    print(solution)

    qubo_quadratic = qubo[0, 1]
    qubo_linear = qubo[0, 0]
    print(f'{qubo_quadratic=}, {qubo_linear=}')
    qubo_ratio = qubo_quadratic / qubo_linear
    print(f'{qubo_ratio=}')

    try:
        # greedy case
        embedding_interaction = solver._solver._register.interactions()[('q0', 'q1')]
    except KeyError:
        # BLaDE case
        embedding_interaction = solver._solver._register.interactions()[(0,1)]

    drive_detuning = solver._solver._drive.detuning._values[-1]
    print(f'{embedding_interaction=}, {drive_detuning=}')
    embedding_drive_ratio = embedding_interaction / drive_detuning
    print(f'{embedding_drive_ratio=}')  # should be 1

    print(f'{-2*qubo_ratio=} should be equal to {embedding_drive_ratio=}')

    return -2*qubo_ratio, embedding_drive_ratio

import pandas as pd

df = pd.DataFrame(columns=['embedder', 'interaction', 'det_ratio', 'drive_ratio'])

for embedder in ['greedy', 'blade']:
    for interaction in [.01, .1, .5, 1, 2, 4, 8]:
        try:
            det_ratio, drive_ratio = embedding_drive_scalings(embedding_method=embedder, interaction=interaction)
            df.loc[len(df.index)] = {
                'embedder': embedder,
                'interaction': interaction,
                'det_ratio': det_ratio,
                'drive_ratio': drive_ratio}
        except Exception as e:
            print(f'Error for {embedder}, {interaction}: {e}')
```

### Tracebacks (optional)

```
   embedder  interaction  det_ratio  drive_ratio
0    greedy         0.10       0.20     0.000020
1    greedy         0.50       1.00     0.000544
2    greedy         1.00       2.00     0.001088
3    greedy         2.00       4.00     0.005158
4    greedy         4.00       8.00     0.010316
5    greedy         8.00      16.00     0.020631
6     blade         0.01       0.02     0.000200
7     blade         0.10       0.20     0.020000
8     blade         0.50       1.00     0.500000
9     blade         1.00       2.00     2.000000
10    blade         2.00       4.00     8.000000
11    blade         4.00       8.00    32.000000
12    blade         8.00      16.00   128.000000
```

### Would you like to work on this issue?

Yes

**Comment by abussy-pasqal** [comment]
Fxed by #223
